packages/uniforms/src/connectField.js

```diff
@@ -44,7 +44,7 @@ export default function connectField (component, {
                     return;
                 }
 
-                if (props.value === undefined && props.required) {
+                if (!props.value && props.required) {
```

Comment:
This should be changed - if a field has a falsy value, like `false` and `initialValue` equal to `true`, it will cause a problem. Empty string is caused by `ensureValue: true` - this flag is used to avoid `undefined` in HTML inputs.

I'd suggest
```diff
- if (props.value === undefined && props.required) {
+ if (props.required && (this.options.ensureValue ? props.value === '' : props.value === undefined)) {
```